stock_picking_show_return/models/stock_picking.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2014-2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2018-2019 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.fields import first
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    returned_ids = fields.Many2many(
+        comodel_name="stock.picking",
+        compute="_compute_returned_ids",
+        string="Returned pickings",
+    )
+    source_picking_id = fields.Many2one(
+        comodel_name="stock.picking",
+        compute="_compute_source_picking_id",
+        string="Source picking",
+    )
+
+    def _compute_returned_ids(self):
+        for picking in self:
+            picking.returned_ids = picking.mapped(
+                "move_ids.returned_move_ids.picking_id"
+            )
+
+    @api.depends("move_ids.origin_returned_move_id")
+    def _compute_source_picking_id(self):
+        """Get source piking from this picking. Only one origin is possible."""
+        for picking in self:
+            picking.source_picking_id = first(
+                picking.mapped("move_ids.origin_returned_move_id.picking_id")
```

Comment:
Mapped not required anymore when the field is a recordset, see https://github.com/odoo/odoo/pull/32897

```suggestion
                picking.move_ids.origin_returned_move_id.picking_id
```

Please apply this in all similar cases